torch/csrc/autograd/engine.cpp

```diff
@@ -864,6 +906,9 @@ auto Engine::compute_dependencies(Node* root, GraphTask& task, uint64_t min_topo
     if (fn->topological_nr() < min_topo_nr) {
       continue;
     }
+    if (!will_use_cuda) {
+      will_use_cuda = fn->stream(c10::DeviceType::CUDA).has_value();
+    }
```

Comment:
@albanD I'm piggybacking on `compute_dependencies` for an advance check if any backward node (including leaf nodes) will use cuda, to decide [if i I need to bother remembering the current streams](https://github.com/pytorch/pytorch/pull/54227/files#diff-c66dfeac2a2da1867233047ec413c7e625644c672d7b38b8ec982f5605923c64R957-R961). Is this the right spot to check if each node has a CUDA stream()? Should i move the check a few lines up, above
```C++
    if (fn->topological_nr() < min_topo_nr) {
      continue;
    }
```?